kartothek/io/eager.py

```diff
@@ -414,8 +436,10 @@ def commit_dataset(
             DeprecationWarning,
         )
 
-    if isinstance(new_partitions, NoDefault):
-        raise TypeError("The parameter `new_partitions` is not optional")
+    if not new_partitions and not metadata and not delete_scope:
+        raise ValueError(
+            "Need to provide either new data, new metadata or a delete scope. None of it was provided."
+        )
```

Comment:
np: do we really care? We aren't making any expensive operations, we could also just log this instead of raising

Reply:
Fair point. I will need to add another test case for this then, to ensure that it actually works and the dataset is not modified at all

Reply:
otoh, "expensive operations" is depending on the perspective. After all, we'll fetch all indices and the metadata from store. this stuff should not be expensive but it is something

Reply:
Right..